Source/ASDisplayNode.mm

```diff
@@ -550,6 +549,14 @@ - (void)_locked_loadViewOrLayer
     _view = [self _locked_viewToLoad];
     _view.asyncdisplaykit_node = self;
     _layer = _view.layer;
+    // Needed to force `-[_ASDisplayLayer setDelegate:]` to evaluate
+    // It appears that UIKit might be setting the delegate ivar directly which
+    // bypasses establishing the `_delegateFlags` ivar on `ASDisplayLayer`
+    // This is critical for things like ASPagerNode, ASCollectionNode that depend
+    // on layer delegateFlags to forward bounds changes
+    if (_layer.delegate == _view) {
+      [_layer setDelegate:_view];
```

Comment:
Being that we don't own the layer, it might be a decent idea locking this to ensure that the `_layer`'s delegate didn't change a microsecond after the if-statement 🤷‍♂